packages/smooth_app/lib/pages/scan/scan_header.dart

```diff
@@ -32,9 +31,11 @@ class _ScanHeaderState extends State<ScanHeader> {
       ),
     );
 
+    final int validBarcodes = model.getAvailableBarcodes().length;
```

Comment:
```suggestion
    final List<String> barcodes = model.getAvailableBarcodes().toList(growable: false);
```

Reasons:
* confusion between "available" and "valid" barcodes
* you call again `model.getAvailableBarcodes()` later in the code
* "validBarcodes" does not sound like a number

If you really want,
```dart
final int numberOfBarcodes = barcodes.length;
```

Reply:
The problem with your `toList()` is that it will force to create a new List in memory.
The Iterable is enough here just to count the number of items.